vcap_utils/vcap_utils/backends/backend_rpc_process.py

```diff
@@ -0,0 +1,204 @@
+        """
+        :param backend_class: The Backend type to initialize
+        :param args: Arguments for the backend
+        :param kwargs: Keyword arguments for the backend
+        """
+        self._backend_class = backend_class
+        self._incoming = multiprocessing.Queue()
+        self._outgoing = multiprocessing.Queue()
+        self._shutdown = multiprocessing.Event()
+        self._futures_lock = RLock()
+        self._futures: Dict[str, Future] = {}
+        """Keep track of ongoing requests in a dict of request_id: Future """
+
+        # Spin up the server process
+        self._process = multiprocessing.Process(
+            target=_rpc_server,
+            daemon=True,
+            name="RpcServer",
+            kwargs={
+                "outgoing": self._incoming,
+                "incoming": self._outgoing,
+                "shutdown": self._shutdown,
+                "backend_class": backend_class,
+                "num_workers": 100,
+                "args": args,
+                "kwargs": kwargs
+            })
+        self._process.start()
+        # Wait for success or failure of the Backend initialization
+        exception = self._incoming.get()
+        if exception:
+            raise exception
+
+        # Start the client thread
+        self._rpc_thread = threading.Thread(
+            target=self._rpc_client,
+            name="RpcClientThread")
+        self._rpc_thread.start()
+
+    @property
+    def workload(self) -> int:
+        # TODO: Use RPC to request the underlying Backends
+        #       'workload' implementation
+        with self._futures_lock:
+            return len(self._futures)
+
+    def _rpc_client(self):
+        """This is the dedicated thread for receiving and routing results
+        from the remote backend. It receives incoming RpcResponses and
+        routes results or exceptions to the relevant Future objects."""
+        while not self._shutdown.is_set() or self._incoming.qsize():
+            try:
+                response: RpcResponse = self._incoming.get(timeout=0.1)
+            except queue.Empty:
+                continue
+            with self._futures_lock:
+                future = self._futures.pop(response.request_id)
+            if response.exception:
+                future.set_exception(response.exception)
+            else:
+                future.set_result(response.result)
+
+    def _rpc_call(self, function: str, *args, **kwargs):
+        """Run a function on the remote backend.
+        :param function: The function to run on the remote backend
+        :param args: Arguments for the function
+        :param kwargs: Keyword arguments for the function
+        """
+        request = RpcRequest(
+            function=function,
+            args=args,
+            kwargs=kwargs,
+            request_id=uuid4().hex
+        )
+
+        future = Future()
+        with self._futures_lock:
+            self._futures[request.request_id] = future
```

Comment:
Is this not thread-safe?

Reply:
Hah, good catch. I added the lock when I was debugging a race condition (that turned out to be caused by something else). 

Either way, it makes me feel better locking the dictionary because it _is_ getting popped by the client thread, it just so happens that it won't be popping that ID since the request hasn't been made yet. 

Should I remove it anyways?

Reply:
you can just add a comment like 
```python
# probably thread safe, but... [your explanation]
```

Reply:
However, `list.pop` is thread safe too

Reply:
I think `dict.__len__` is too